tests/std/tests/P0009R18_mdspan_layout_stride/test.cpp

```diff
@@ -400,22 +400,85 @@ constexpr void check_required_span_size() {
+        assert(m.is_exhaustive() == expected);
+    };
+
+    // rank() is equal to 0
+    check(extents<int>{}, array<int, 0>{}, true);
+
+    // rank() is equal to 1
+    check(extents<int, 0>{}, array{1}, true);
+    check(dextents<int, 1>{0}, array{2}, false);
+    check(extents<int, 1>{}, array{3}, false);
+    check(dextents<int, 1>{2}, array{2}, false);
+    check(extents<int, 3>{}, array{1}, true);
+    check(dextents<int, 1>{4}, array{1}, true);
+
+    // rank() is equal to 2
+    check(extents<int, 3, 3>{}, array{1, 3}, true);
+    check(extents<int, dynamic_extent, 3>{3}, array{3, 1}, true);
+    check(extents<int, 3, dynamic_extent>{3}, array{4, 1}, false);
+    check(dextents<int, 2>{3, 3}, array{3, 1}, true);
+    check(extents<int, 4, 5>{}, array{5, 1}, true);
+    check(extents<int, 6, dynamic_extent>{5}, array{1, 6}, true);
+    check(extents<int, dynamic_extent, 7>{5}, array{1, 8}, false);
+    check(dextents<int, 2>{6, 5}, array{1, 10}, false);
+    check(extents<int, 0, 3>{}, array{3, 1}, true);
+    check(extents<int, dynamic_extent, 3>{0}, array{6, 1}, false);
+    check(extents<int, 0, dynamic_extent>{3}, array{6, 2}, false);
+    check(dextents<int, 2>{0, 3}, array{7, 2}, false);
+    check(extents<int, 0, 0>{}, array{1, 1}, false);
+    check(extents<int, 0, dynamic_extent>{0, 0}, array{1, 1}, false);
+    check(dextents<int, 2>{0, 0}, array{1, 2}, false);
+    check(extents<int, 1, dynamic_extent>{0}, array{1, 2}, false);
+
+    // rank() is greater than 2
+    check(extents<int, 2, 3, 5>{}, array{1, 2, 6}, true);
+    check(extents<int, dynamic_extent, 3, 5>{2}, array{1, 10, 2}, true);
+    check(extents<int, 2, 3, dynamic_extent>{5}, array{3, 1, 6}, true);
+    check(extents<int, dynamic_extent, dynamic_extent, 5>{2, 3}, array{15, 1, 3}, true);
+    check(extents<int, 2, dynamic_extent, dynamic_extent>{3, 5}, array{5, 10, 1}, true);
+    check(dextents<int, 3>{2, 3, 5}, array{15, 5, 1}, true);
+    check(extents<int, 2, 5, 8>{}, array{1, 2, 12}, false);
+    check(extents<int, 2, dynamic_extent, 8>{5}, array{8, 18, 1}, false);
+    check(dextents<int, 3>{2, 5, 8}, array{5, 1, 12}, false);
+
+    // rank() is greater than 2 and some extents are equal to 0
+    check(extents<int, 2, 0, 7>{}, array{7, 14, 1}, true);
+    check(extents<int, dynamic_extent, 0, 7>{2}, array{1, 14, 2}, true);
+    check(extents<int, 2, dynamic_extent, 7>{0}, array{14, 28, 1}, false);
+    check(extents<int, 2, dynamic_extent, dynamic_extent>{0, 7}, array{1, 2, 2}, false);
+    check(dextents<int, 3>{2, 0, 7}, array{2, 28, 4}, false);
+    check(extents<int, 5, 0, 0>{}, array{3, 1, 1}, false);
+    check(extents<int, 5, dynamic_extent, 0>{0}, array{1, 5, 1}, false);
+    check(dextents<int, 3>{5, 0, 0}, array{2, 1, 10}, false);
+    check(extents<int, 0, 0, 0>{}, array{1, 1, 1}, false);
+    check(extents<int, 0, 1, 1>{}, array{1, 1, 1}, true);
+    check(std::extents<int, 0, 3>{}, std::array{6, 2}, false);

```

Comment:
Unnecessary qualification. But more significantly, this doesn't follow "rank() is greater than 2". I'll group it with the other `check(extents<int, 0, 3>{}`.